mobile/library/common/jni/jni_interface.cc

```diff
@@ -306,19 +308,18 @@ static void pass_headers(const char* method, envoy_headers headers, jobject j_co
   }
 
   env->DeleteLocalRef(jcls_JvmCallbackContext);
-  release_envoy_headers(headers);
 }
 
 // Platform callback implementation
 // These methods call jvm methods which means the local references created will not be
 // released automatically. Manual bookkeeping is required for these methods.
 
-static void* jvm_on_headers(const char* method, envoy_headers headers, bool end_stream,
-                            envoy_stream_intel stream_intel, void* context) {
+static void* jvm_on_headers(const char* method, const Envoy::Types::ManagedEnvoyHeaders& headers,
```

Comment:
this one isn't called by jvm directly either.  optional nit onHeadersFromJvm or something

Reply:
it's calling a method from the java world tho which I think is why it has a `jvm` prefix (`pass_headers - now renamed to `passHeaders` - method did not have `jvm` prefix as it was/is not calling into `jvm`). 

I do not have a strong opinion in here but will keep the things they are until we figure out new naming patterns for this file - ideally in a separate PR.